pkg/bootstrap/render.go

```diff
@@ -19,214 +19,183 @@ import (
-	"manifests/klusterlet/clusterrole_priority_class.yaml",
-	"manifests/klusterlet/priority_class.yaml",
-}
-
-type RenderConfig struct {
-	KlusterletRenderConfig
-	ImagePullSecretConfig
 }
 
 type BootstrapKubeConfigSecret struct {
 	Name       string
 	KubeConfig string
 }
 
-// KlusterletRenderConfig defines variables used in the klusterletFiles.
-type KlusterletRenderConfig struct {
-	KlusterletName            string
-	KlusterletNamespace       string
-	ManagedClusterNamespace   string
-	RegistrationOperatorImage string
-	RegistrationImageName     string
-	WorkImageName             string
-	ImageName                 string
-	PriorityClassName         string
-	InstallMode               string
-
-	NodeSelector  map[string]string
-	Tolerations   []corev1.Toleration
-	NodePlacement *operatorv1.NodePlacement
-
-	RegistrationConfiguration *operatorv1.RegistrationConfiguration
-	WorkConfiguration         *operatorv1.WorkAgentConfiguration
-
-	MultipleHubsEnabled              bool
-	DefaultBootstrapKubeConfigSecret BootstrapKubeConfigSecret
-	BootstrapKubeConfigSecrets       []BootstrapKubeConfigSecret
+type KlusterletManifestsConfig struct {
+	chartConfig      *chart.KlusterletChartConfig
+	managedCluster   *clusterv1.ManagedCluster
+	klusterletConfig *klusterletconfigv1alpha1.KlusterletConfig
 }
 
-type ImagePullSecretConfig struct {
-	UseImagePullSecret       bool
-	ImagePullSecretName      string
-	ImagePullSecretData      string
-	ImagePullSecretConfigKey string
-	ImagePullSecretType      corev1.SecretType
+func NewKlusterletManifestsConfig(installMode operatorv1.InstallMode,
+	clusterName string, bootstrapKubeConfig []byte) *KlusterletManifestsConfig {
+	return &KlusterletManifestsConfig{
+		chartConfig: newKlusterletChartConfig(installMode, clusterName, bootstrapKubeConfig),
+	}
 }
 
-type KlusterletManifestsConfig struct {
-	InstallMode operatorv1.InstallMode
-
-	ClusterName                  string
-	KlusterletNamespace          string
-	KlusterletClusterAnnotations map[string]string
-	BootstrapKubeconfig          []byte
-
-	// PriorityClassName is the name of the PriorityClass used by the klusterlet and operator
-	PriorityClassName string
-
-	// Used to determine whether mc is a localcluster.
-	ManagedCluster *clusterv1.ManagedCluster
-
-	klusterletconfig *klusterletconfigv1alpha1.KlusterletConfig
+func newKlusterletChartConfig(installMode operatorv1.InstallMode,
+	clusterName string, bootstrapKubeConfig []byte) *chart.KlusterletChartConfig {
+	allowPrivilegeEscalation := false
+	privileged := false
+	runAsNonRoot := true
+	readOnlyRootFilesystem := true
+
+	chartConfig := &chart.KlusterletChartConfig{
+		ReplicaCount: 1,
+		Images: chart.ImagesConfig{
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			ImageCredentials: chart.ImageCredentials{
+				// set true by default
+				CreateImageCredentials: true,
+			},
+		},
+		// the release ns and the agent ns are same, so disable to generate ns.
+		CreateNamespace: true,
```

Comment:
the comment said "disable to generate ns", but the coce is true, please confirm.

Reply:
CreateNamespace should be true. removed the comment.